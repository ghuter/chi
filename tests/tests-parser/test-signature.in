A :: signature {
	f :: fun();
}

B :: signature [T] {
	f :: fun() ^T;
}

C :: signature [K, V] {
	f :: fun(k: ^K) ^T;
}

D :: signature [K, V] (X: Module) {
	f :: fun (k: ^K) ^V;
}

E :: signature [K, V] (X: Module[K]) {
	f :: fun (k: ^K) ^V;
}

F :: signature [K, V] (X: Module[K], Y: Other[K, V]) {
	f :: fun (k: ^K) ^V;
}

G :: signature (X: Module, Y: Other) {
	f :: fun (k: i32) i32;
}

G :: signature (X: Module) {
	f :: fun (k: i32) i32;
}
