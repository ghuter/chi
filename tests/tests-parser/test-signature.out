SMODSIGN(A, [], (), {SSIGN(f) : void})
SMODSIGN(B, [T], (), {SSIGN(f) : ^T})
SMODSIGN(C, [K, V], (), {SSIGN(f, k : ^K) : ^T})
SMODSIGN(D, [K, V], (X: Module[]), {SSIGN(f, k : ^K) : ^V})
SMODSIGN(E, [K, V], (X: Module[K]), {SSIGN(f, k : ^K) : ^V})
SMODSIGN(F, [K, V], (X: Module[K], Y: Other[K, V]), {SSIGN(f, k : ^K) : ^V})
SMODSIGN(G, [], (X: Module[], Y: Other[]), {SSIGN(f, k : i32) : i32})
SMODSIGN(G, [], (X: Module[]), {SSIGN(f, k : i32) : i32})
